### ISSUE TITLE
Does it need java to run?

### ISSUE THREAD

**Issue by gaurav517** [bcaudan/node-sonar-scanner#13]
Hello,

Does it need JRE to run the analysis? I have pure javascript project and I don't want to install Java just for this purpose. What are options?

Thanks,
Kumar

**Comment by brunobastosg** [comment]
Yes, it does. I tested it locally and it worked, but I had Java installed. Once I tried to run it in a Node docker image, it didn't work. I had to use a Node + Java docker image.

**Comment by kirill-konshin** [comment]
This just blows my mind...

It reveals that simple uploader does too many things. It also requires `typescript` locally if you want to deal with TS files. According to docs https://docs.sonarqube.org/latest/analysis/languages/typescript/ you must `npm install typescript` so you must have Node and must have Java. Which makes it impossible to do nice clean things like build & test & gather coverage using Node image and upload using Java image. 

And yes, the only solution is to create a Node+Java Docker.
